tests/test_utils/extend_tuple.rs

```diff
@@ -0,0 +1,177 @@
+            to_extend.1,
+            to_extend.2,
+            to_extend.3,
+            to_extend.4,
+            to_extend.5,
+            self,
+        )
+    }
+}
+
+impl<T1, T2, T3, T4, T5, T6, T7, T8> ExtendTuple<(T1, T2, T3, T4, T5, T6, T7)> for T8 {
+    type Output = (T1, T2, T3, T4, T5, T6, T7, T8);
+
+    fn extend_tuple(self, to_extend: (T1, T2, T3, T4, T5, T6, T7)) -> Self::Output {
+        (
+            to_extend.0,
+            to_extend.1,
+            to_extend.2,
+            to_extend.3,
+            to_extend.4,
+            to_extend.5,
+            to_extend.6,
+            self,
+        )
+    }
+}
+
+impl<T1, T2, T3, T4, T5, T6, T7, T8, T9> ExtendTuple<(T1, T2, T3, T4, T5, T6, T7, T8)> for T9 {
+    type Output = (T1, T2, T3, T4, T5, T6, T7, T8, T9);
+
+    fn extend_tuple(self, to_extend: (T1, T2, T3, T4, T5, T6, T7, T8)) -> Self::Output {
+        (
+            to_extend.0,
+            to_extend.1,
+            to_extend.2,
+            to_extend.3,
+            to_extend.4,
+            to_extend.5,
+            to_extend.6,
+            to_extend.7,
+            self,
+        )
+    }
+}
+
+impl<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10> ExtendTuple<(T1, T2, T3, T4, T5, T6, T7, T8, T9)>
+    for T10
+{
+    type Output = (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10);
+
+    fn extend_tuple(self, to_extend: (T1, T2, T3, T4, T5, T6, T7, T8, T9)) -> Self::Output {
+        (
+            to_extend.0,
+            to_extend.1,
+            to_extend.2,
+            to_extend.3,
+            to_extend.4,
+            to_extend.5,
+            to_extend.6,
+            to_extend.7,
+            to_extend.8,
+            self,
+        )
+    }
+}
+
+impl<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11>
+    ExtendTuple<(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10)> for T11
+{
+    type Output = (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11);
+
+    fn extend_tuple(self, to_extend: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10)) -> Self::Output {
+        (
+            to_extend.0,
+            to_extend.1,
+            to_extend.2,
+            to_extend.3,
+            to_extend.4,
+            to_extend.5,
+            to_extend.6,
+            to_extend.7,
+            to_extend.8,
+            to_extend.9,
+            self,
+        )
+    }
+}
+
+impl<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12>
+    ExtendTuple<(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11)> for T12
+{
+    type Output = (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12);
+
+    fn extend_tuple(
+        self,
+        to_extend: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11),
+    ) -> Self::Output {
+        (
+            to_extend.0,
+            to_extend.1,
+            to_extend.2,
+            to_extend.3,
+            to_extend.4,
+            to_extend.5,
+            to_extend.6,
+            to_extend.7,
+            to_extend.8,
+            to_extend.9,
+            to_extend.10,
+            self,
+        )
+    }
+}
```

Comment:
This is, in my opinion, worse than a bunch of nested for loops... But I wonldn't prevent merging over that. I think this PR is ok.

Reply:
Yeah, these are quite ugly - I can wrap them in macros to remove at least the redundancy.

Reply:
I don't feel like that would be worth it. A few nested for loops are better than : 
 - adding a new dependency
 - creating a new trait that will be used just in one test case
 - creating a new trait AND a macro to implement that trait

We are talking about a single test. No need to overcomplicate things.

Even the principle of testing all possible combinations of syntaxes sounds a bit overkill to me.

I don't think we should delay merging that PR for longer. I would have preferred simpler tests, but the current state is okay. We can merge this now, I think.

@alamb

Reply:
Ok!

Reply:
I'm not in favor of merging this (I think its too much machinery/code for a test case). The previous version with the combinatorial for loops I'd rather not either but I don't feel as strongly against merging that one at least so I'm defo OK with doing that if everyone's fine with it. (doesn't mean we can't merge this either ofc, merely my preference)

Reply:
I have replaced the complete CreateTrigger test suite with some selected test cases in commit [459dae1](https://github.com/sqlparser-rs/sqlparser-rs/pull/1352/commits/459dae1415590f301e9ab272f426bb8ff5140d4e), as I believe the request was. I am not sure whether this was what was requested.